site/source/docs/tools_reference/emcc.rst

```diff
@@ -305,9 +305,9 @@ Options that are modified or new in *emcc* are listed below:
 
 ``--post-js <file>``
   [link]
-  Like ``--pre-js``, but emits a file *after* the emitted code. In addition to
-  this code being located after the emitted JS library code, this code will also
-  be executed after main() has run.
+  Like "--pre-js", but emits a file *after* the emitted code. In addition to
+  being located after the emitted JS library code, --post-js code will also
+  execute only after main() has run.
```

Comment:
Is this actually the behavior we want though? 

Is this true today?  I would have though that `WASM_ASYNC_COMPILATION=1` (which is the default) means that the code runs before we run main (while the wasm file is loading).

Which behaviour do we think is more useful / expected?

Reply:
I guess maybe we should be consistent, but for now it seems like `-sMODULARIZE` has already been updated to have the well-defined behaviour of running post-js after main.

For code built without `-sMODULARIZE` it currently depends if you build with async compilation or not, so its not as simple as saying that post-js code always runs after main.

So, perhaps revert the documentation change here until and we can separately consider how to document the current differences between modes (or remove them and then document the consistent new behaviour)?

Documenting this as always being one way would currently be inaccurate I think?